### PR TITLE
[Easy] Cleanup Balancer Module

### PR DESCRIPTION
**Pull request by bh2smith**
- Rename `PoolCacheMetrics` to `WeightedPoolCacheMetrics` to avoid possibility of confusing invalid imports 
- A few useful log statements in event handler

Note that the last item may be removed upon suggestion (since its not yet used and needs some rework).

### Test Plan
no logic changes.

### CODE REVIEW

File: shared/src/balancer/pool_cache.rs [hunk 1]
```diff
@@ -13,10 +14,6 @@ use ethcontract::errors::MethodError;
 use ethcontract::{BlockId, Bytes, H160, H256, U256};
 use std::{collections::HashSet, sync::Arc};
 
-pub trait PoolCacheMetrics: Send + Sync {
```
Comment (nlordell) [inline shared/src/balancer/pool_cache.rs:16]: The idea with this extra indirection is we can have separate metrics for Balancer vs Uniswap pool fetching. It might be worth leaving this in.

Alternatively we can just make the metrics implement `CacheMetrics` directly and avoid the extra level of indirection with the `PoolCacheMetrics` trait.

File: shared/src/balancer/pool_storage.rs [hunk 2]
```diff
@@ -192,6 +219,11 @@ impl PoolStorage {
                     .insert(pool_id);
             }
         }
+        tracing::info!(
+            "inserted {} balancer pool creation events. Synced to block {}",
+            num_events,
+            self.last_event_block()
```
Comment (nlordell) [inline shared/src/balancer/pool_storage.rs:225]: Is this accurate? If we sync to block X but the last pool was created at some other block Y won't this log be misleading?

Comment (bh2smith) [reply shared/src/balancer/pool_storage.rs:225]: Yes... although it does say up till which block we have pools. If this is too misleading or confusing, I would rather just remove it.

File: shared/src/balancer/pool_cache.rs (hunk 1 above)
Comment (bh2smith) [reply shared/src/balancer/pool_cache.rs:16]: I don't really have a preference here. I would lean toward avoid extra direction, but I feel like this would be an unrelated change to this PR.

File: shared/src/balancer/pool_storage.rs (hunk 2 above)
Comment (nlordell) [reply shared/src/balancer/pool_storage.rs]: It says `Synced to block` which is not true, since the latest pool we have is not necessarily on the latest block we've synced.

File: shared/src/balancer/pool_cache.rs (hunk 1 above)
Comment (nlordell) [reply shared/src/balancer/pool_cache.rs:16]: Let's wait and see what others think, but IMO it is related to this PR as we are removing one of the traits here, so we should either remove both `PoolCacheMetrics` in order to remove one level of indirection, or keep the extra indirection and allow separate metrics to be used.

Comment (vkgnosis) [reply shared/src/balancer/pool_cache.rs:16]: I feel that it is useful to have separate metrics for separate pool fetching backends. Not very important, just slightly in favor.

**Review by vkgnosis — APPROVED** [review]

File: shared/src/balancer/pool_cache.rs (hunk 1 above)
Comment (bh2smith) [reply shared/src/balancer/pool_cache.rs:16]: The reason I removed this is because there is a bit of a name conflict if we use `PoolCacheMetrics` because its easy to import the wrong one. I could also get around this by putting it back and calling it `WeightedPoolCacheMetrics`. I will do that for now.

Comment (bh2smith) [reply shared/src/balancer/pool_cache.rs:16]: Cool. then I will keep them all separated, but rename this one to `WeightedPoolCacheMetrics` so that imports are not easily confused.

File: shared/src/balancer/pool_storage.rs (hunk 2 above)
Comment (bh2smith) [reply shared/src/balancer/pool_storage.rs]: Yea I see the problem. This log has been removed.

File: shared/src/balancer/pool_cache.rs [hunk 3]
```diff
@@ -3,7 +3,7 @@ use crate::balancer::pool_storage::RegisteredWeightedPool;
 use crate::pool_fetching::{handle_contract_error, MAX_BATCH_SIZE};
 use crate::{
     balancer::pool_storage::WeightedPool,
-    recent_block_cache::{Block, CacheFetching, CacheKey, CacheMetrics, RecentBlockCache},
+    recent_block_cache::{Block, CacheFetching, CacheKey, RecentBlockCache},
```
Comment (nlordell) [inline shared/src/balancer/pool_cache.rs:6]: nit: This is an unnecessary change - it looks like the `use` statement was just moved to a different line.

**Review by nlordell — APPROVED** [review]

File: shared/src/balancer/pool_cache.rs (hunk 3 above)
Comment (bh2smith) [reply shared/src/balancer/pool_cache.rs]: Yea, I just went over all the import statements and consolidated them. So there are a few more changed imports but now they are all in the most compact form throughout the module.
